ledger/simulation/tracer.go

```diff
@@ -178,18 +250,40 @@ func (tracer *evalTracer) saveEvalDelta(evalDelta transactions.EvalDelta, appIDT
 	applyDataOfCurrentTxn.EvalDelta.InnerTxns = inners
 }
 
+func (tracer *evalTracer) makeOpcodeTraceUnit(cx *logic.EvalContext) OpcodeTraceUnit {
+	return OpcodeTraceUnit{PC: uint64(cx.PC())}
+}
+
 func (tracer *evalTracer) BeforeOpcode(cx *logic.EvalContext) {
+	currentOpcodeUnit := tracer.makeOpcodeTraceUnit(cx)
+
+	// logic sig opcode part
 	if cx.RunMode() != logic.ModeApp {
 		// do nothing for LogicSig ops
+		if tracer.result.ExecTraceConfig == NoExecTrace {
+			return
+		}
+		// BeforeOpcode runs for logic sig happens before txn group exec, including app calls
+		// get cx.GroupIndex() and append to trace
+		indexIntoTxnGroup := cx.GroupIndex()
+		execTrace := tracer.result.TxnGroups[0].Txns[indexIntoTxnGroup].Trace
```

Comment:
I haven't digested what's going on yet, but it seems very surprising to see explicit 0 indexes.  Why is that?

Reply:
We are only supporting 1 txn-group simulation at this point, so there are a few places of explicit 0 indices.

Reply:
Oh, I was reading it as the first txn in the group, not the first group. Thanks.